.github/workflows/check-available-os-versions.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/sh
+
+set -e
+
+SURPRISE_COUNT=0
+[ -z "${HTML2TEXT}" ] && HTML2TEXT=html2text
+
+fetch_webpage_as_markdown() {
+	curl -sS "$1" | ${HTML2TEXT}
+}
+
+fetch_alpine_upstream_latest_stable() {
+	fetch_webpage_as_markdown 'https://alpinelinux.org/downloads/' \
+		| grep 'Current Alpine Version' \
+		| awk '{print $4}' \
+		| sed -e 's|*||g'
+}
+
+fetch_github_runner_list() {
+	fetch_webpage_as_markdown 'https://github.com/actions/runner-images' \
+		| perl -ne 'next unless /YAML Label/; print <>;' \
+		| perl -ne 'next if /^---/; print;' \
+		| perl -ne 'last if /Label scheme/; print;' \
+		| grep -v ' beta ' \
+		| grep -v ' deprecated ' \
+		| grep -v 'windows-' \
+		| cut -d'|' -f2 \
+		| sed -e 's| or |\n|g' -e 's|, |\n|g' -e 's|^ ||' -e 's|`||g' -e 's| $||' \
+		| grep -v -- '-latest' \
+		| grep -v -- '-large' \
+		| grep -v -- '-xlarge' \
+		| grep -v '^\s*$'
+}
+
+fetch_github_runner_os_versions() {
+	osname="$1"
+	osname_lower="$(echo ${osname} | tr '[:upper:]' '[:lower:]')"
+	fetch_github_runner_list \
+		| grep "^${osname_lower}-" \
+		| sed -e "s|^${osname_lower}-||" \
+		| tr '\n' ' ' \
+		| sed -e 's| $||'
+
+}
+
+fetch_vmactions_os_versions() {
+	osname="$1"
+	osname_lower="$(echo ${osname} | tr '[:upper:]' '[:lower:]')"
+	fetch_webpage_as_markdown "https://github.com/vmactions/${osname_lower}-vm" \
+		| grep 'All the supported' \
+		| sed -e "s|.*${osname} ||i" -e 's|, test\.releases.*||g' -e 's|,||g' \
+		| tr ' ' '\n' \
+		| sort -rn \
+		| tr '\n' ' ' \
+		| sed -e 's| $||'
+}
+
+fetch_os_versions() {
+	osname="$1"
+	case ${osname} in
+	Alpine)
+		fetch_alpine_upstream_latest_stable
+		;;
+	Fedora)
+		echo rawhide
+		;;
+	*BSD|Solaris)
+		fetch_vmactions_os_versions $1
+		;;
+	macOS|Ubuntu)
+		fetch_github_runner_os_versions $1
+		;;
+	*)
+		echo unknown
+	esac
+}
+
+check_os_versions() {
+	osname="$1"; shift
+	known="$@"
+	available="$(fetch_os_versions ${osname})"
+	[ "${known}" = "${available}" ] || {
+		SURPRISE_COUNT=$(expr 1 + ${SURPRISE_COUNT})
+		echo "${osname}:"
+		echo "-${known}"
+		echo "+${available}"
+	}
+}
+
+main() {
+	# TODO:
+	# check that all platforms in platform-builders.yml are being checked
+
+	check_os_versions Alpine 3.20.0
+	check_os_versions DragonFlyBSD 6.4.0
+	check_os_versions Fedora rawhide
+	check_os_versions FreeBSD 14.0 13.3 13.2 12.4
+	check_os_versions macOS 14 13 12
+	check_os_versions NetBSD 10.0 9.4 9.3 9.2 9.1 9.0
+	check_os_versions OpenBSD 7.5 7.4 7.3 7.2
+	check_os_versions Solaris 11.4-gcc 11.4
+	check_os_versions Ubuntu 22.04 20.04
```

Comment:
Very nice and concise config-file like syntax, which I use a lot for scripts.
Would it be worth adding a comment on each .yaml file to tell "keep up to date with `check-available-os-versions.sh`"?